hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -158,56 +158,40 @@ case class EmitRegion(mb: EmitMethodBuilder[_], region: Value[Region]) {
 }
 
 object EmitValue {
-  def apply(missing: Option[Value[Boolean]], v: SValue): EmitValue = new EmitValue {
-    lazy val required: Boolean = missing match {
-      case None => true
-      case Some(m) => Code.constBoolValue(m).contains(false)
-    }
+  def apply(missing: Option[Value[Boolean]], v: SValue): EmitValue = new EmitValue(missing, v)
+}
 
-    override lazy val emitType: EmitType = EmitType(v.st, required)
+class EmitValue(missing: Option[Value[Boolean]], v: SValue) {
+  lazy val required: Boolean = missing match {
+    case None => true
+    case Some(m) => Code.constBoolValue(m).contains(false)
+  }
 
-    override def load: EmitCode = {
-      val ec = EmitCode(Code._empty,
-        if (required) const(false) else missing.get,
-        v)
-      assert(ec.required == required)
-      ec
-    }
+  lazy val emitType: EmitType = EmitType(v.st, required)
 
-    override def get(cb: EmitCodeBuilder): SCode = {
-      if (required) {
-        v
-      } else {
-        cb.ifx(missing.get, cb._fatal(s"Can't convert missing ${ st } to PValue"))
-        v
-      }
+  def valueTuple(): IndexedSeq[Value[_]] = {
+    missing match {
+      case Some(m) => v.valueTuple :+ m
+      case None => v.valueTuple
     }
   }
-}
-
-abstract class EmitValue {
-  def emitType: EmitType
-
-  def st: SType = emitType.st
-
-  def load: EmitCode
-
-  def get(cb: EmitCodeBuilder): SCode
-}
-
-class EmitUnrealizableValue(private val ec: EmitCode) extends EmitValue {
-  val emitType: EmitType = ec.emitType
-
-  assert(st.isInstanceOf[SStream])
-  private[this] var used: Boolean = false
 
   def load: EmitCode = {
-    assert(!used)
-    used = true
+    val ec = EmitCode(Code._empty,
+      if (required) const(false) else missing.get,
+      v)
+    assert(ec.required == required)
     ec
   }
 
-  override def get(cb: EmitCodeBuilder): SCode = throw new UnsupportedOperationException(s"Can't make PValue for unrealizable type ${ ec.st }")
+  def get(cb: EmitCodeBuilder): SCode = {
```

Comment:
get could return a value instead of a code, I think? I know that could happen in a future PR.

Reply:
Turns out `EmitValue.get` only has one use!